test/mpi/configure.ac

```diff
@@ -641,6 +641,8 @@ AC_PROG_CC
 AC_PROG_CC_C99
 AM_PROG_CC_C_O
 
+AC_USE_SYSTEM_EXTENSIONS
```

Comment:
This is a pretty heavy hammer for strdup usage. Unfortunately our solutions for strdup are pretty inconsistent across the repo. We have `MPL_strdup` for portability, but still some `strdup` calls have slipped in. And of course we have no MPL in the testsuite. I guess I favor importing a fallback version, or using the `PAC_FUNC_NEEDS_DECL` (like dtpools does) over adding these extensions.

Reply:
Tests have a different focus from the mpich library. With tests, we would like to keep it as simple as we can so we don't have to spend extra effort maintaining the extra complexity in tests and less effort in debugging. I think the "heavy hammer" solution that allows tests to simply use functions that supported by common UNIXes is simpler than adding dependency on MPL or maintaining the extra fallbacks + configure logics. What's the con side of using `AC_USE_SYSTEM_EXTENSIONS`?

Reply:
Tests gradually become less portable.

Reply:
> 
> 
> Tests gradually become less portable.

Yeah. Portablity and Complexity often are conflicting goals. I think with tests, we want to lean toward the simple side (rather than the portable side -- as long as we don't encounter issues). We always can add the `strdup` fallback -- or rather, avoid it altogether -- when we encounter the first case when it is not not natively supported (and `AC_USE_SYSTEM_EXTENSIONS` failed to help).

Reply:
If this were just the internal testsuite for MPICH, I would agree. However, the MPICH testsuite is the most comprehensive publicly available MPI testsuite. I think portability is still important to maintain.

Reply:
Commits modified to directly check `strdup` (and `usleep`) in stead of using `AC_USE_SYSTEM_EXTENSIONS`